src/mono/dbi/CMakeLists.txt

```diff
@@ -0,0 +1,151 @@
+    elseif(HOST_ARM64)
+        set(CLR_CMAKE_HOST_ARCH arm64)
+    elseif(HOST_ARM)
+        set(CLR_CMAKE_HOST_ARCH arm)
+    elseif(HOST_AMD64)
+        set(CLR_CMAKE_HOST_ARCH x64)
+    endif()
+endif()
+
+add_definitions(-DDBI_COMPONENT_MONO)
+
+include_directories(
+  ${CMAKE_CURRENT_SOURCE_DIR}/../..
+  ${PROJECT_SOURCE_DIR}/../
+  ${PROJECT_SOURCE_DIR}/../dbi
+  ${PROJECT_SOURCE_DIR}/../dbi/socket-dbi
+  ${PROJECT_SOURCE_DIR}/../../coreclr/md/enc
+  ${PROJECT_SOURCE_DIR}/../../coreclr/inc
+  ${PROJECT_SOURCE_DIR}/../../coreclr/md/inc
+  ${PROJECT_SOURCE_DIR}/../../coreclr/md/compiler)
+
+set(mscorbi_sources_base
+    cordb.cpp
+    cordb.h
+    cordb-appdomain.cpp
+    cordb-appdomain.h
+    cordb-assembly.cpp
+    cordb-assembly.h
+    cordb-blocking-obj.cpp
+    cordb-blocking-obj.h
+    cordb-breakpoint.cpp
+    cordb-breakpoint.h
+    cordb-chain.cpp
+    cordb-chain.h
+    cordb-class.cpp
+    cordb-class.h
+    cordb-code.cpp
+    cordb-code.h
+    cordb-eval.cpp
+    cordb-eval.h
+    cordb-frame.cpp
+    cordb-frame.h
+    cordb-function.cpp
+    cordb-function.h
+    cordb-process.cpp
+    cordb-process.h
+    cordb-register.cpp
+    cordb-register.h
+    cordb-stepper.cpp
+    cordb-stepper.h
+    cordb-thread.cpp
+    cordb-thread.h
+    cordb-type.cpp
+    cordb-type.h
+    cordb-value.cpp
+    cordb-value.h
+)
+
+
+if(HOST_DARWIN)
+set(OS_LIBS "-framework CoreFoundation" "-framework Foundation")
+elseif(HOST_LINUX)
+set(OS_LIBS pthread m dl)
+elseif(HOST_WIN32)
+set(OS_LIBS bcrypt.lib Mswsock.lib ws2_32.lib psapi.lib version.lib advapi32.lib winmm.lib kernel32.lib)
+endif()
+
+addprefix(mscorbi_sources ../dbi/ "${mscorbi_sources_base}")
+add_subdirectory(${PROJECT_SOURCE_DIR}/socket-dbi)
+
+include(${PROJECT_SOURCE_DIR}/../../../eng/native/configuretools.cmake)
+include(${PROJECT_SOURCE_DIR}/../../../eng/native/configurepaths.cmake)
+include(${PROJECT_SOURCE_DIR}/../../../eng/native/configureplatform.cmake)
+include(${PROJECT_SOURCE_DIR}/../../../eng/native/configurecompiler.cmake)
+
+if (CLR_CMAKE_HOST_UNIX)
+  include_directories("${PROJECT_SOURCE_DIR}/../../coreclr/pal/inc")
+  include_directories("${PROJECT_SOURCE_DIR}/../../coreclr/pal/inc/rt")
+  include_directories("${PROJECT_SOURCE_DIR}/../../coreclr/pal/src/safecrt")
+
+  append("-Wno-missing-prototypes  -Wno-pointer-arith -Wno-macro-redefined" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
+  add_subdirectory(${PROJECT_SOURCE_DIR}/../../coreclr/pal pal)
+
+  include_directories("../../coreclr/pal/inc/rt/cpp")
+  add_compile_options(-nostdinc)
+endif (CLR_CMAKE_HOST_UNIX)
+
+include_directories("../../coreclr/pal/prebuilt/inc")
+include_directories("../../coreclr/nativeresources")
+
+if (CLR_CMAKE_HOST_UNIX)
+    add_subdirectory(${PROJECT_SOURCE_DIR}/../../coreclr/nativeresources nativeresources)
+endif()
+
+add_subdirectory(${PROJECT_SOURCE_DIR}/../../coreclr/md/runtime md/runtime)
+add_subdirectory(${PROJECT_SOURCE_DIR}/../../coreclr/md/compiler md/compiler)
```

Comment:
In my opinion this would be permanent.
@lambdageek, do you think we should move directories that I'm using to "native" directory?

Reply:
This is a permanent dependency.

I think it would be good to componentize this part of CoreCLR.  I'm not sure whether moving this to src/native is worth it. It's still more closely tied to CoreCLR than to Mono.  (Mono itself doesn't use it.  Just mono's cordb).

Maybe moving some files around under src/coreclr/md to more clearly define what is a separable component here makes sense.

Certainly we should teach our CI to run debugger tests on mono (or at least build Mono's cordb when those files change).

Reply:
I don't have a strong opinion as most of the developers I work with day to day rarely make changes in these areas of the code. Adding @davidwrighton or @jkotas who might be better positioned to represent devs that touch that code more frequently in case they have any concerns.

Reply:
I expect that we will eventually end up in a place where majority of the unmanaged code is shared between CoreCLR and Mono, similar how vast majority of managed code is shared between CoreCLR and Mono today. 

I think it is good idea to be moving the shared pieces under `src/native` to make it obvious what is shared.

Reply:
@lambdageek should move files in this PR or in another one?

Reply:
My personal preference is to separate re-organization PRs from PRs that add functionality.  Also this PR is already pretty big. I would do it in a follow-up PR.

Reply:
I agree that it is a good idea to do the move in separate PR.